### PR TITLE
Spec exception handling v1.0 rc1

### PR DESCRIPTION
**Pull request by killenb**

### CODE REVIEW

File: doc/spec_exceptionHandling.dox [hunk 1]
```diff
@@ -52,13 +55,16 @@ namespace ChimeraTK {
...
     - 2.3.3 The return value of write() indicates whether data was lost in the transfer. If the write has to be delayed due to an exception, the return value will be true (= data lost) if a previously delayed and not-yet written value is discarded in the process, false (= no data lost) otherwise.
     - \anchor b_2_3_4 2.3.4 When the delayed value is finally written to the device during the recovery procedure, the return value of the write() is ignored. \ref comment_b_2_3_4 "(*)"
     - 2.3.5 It is guaranteed that the write takes place before the device is considered fully recovered again and other transfers are allowed (cf. \ref b_3_1 "3.1").
   - \anchor b_2_4 2.4 In case of exceptions, there is no guaranteed realtime behaviour, not even for "non-blocking" transfers. \ref comment_b_2_4 "(*)"
+  - \anchor b_2_5 2.5 TransferElement::isReadable(), TransferElement::isWriteable() and TransferElement::isReadonly() return with values as if reading and writing would be allowed. \ref comment_b_2_5 "(*)"
```
[the remark this answers is not in the document]
Comment (killenb) [reply doc/spec_exceptionHandling.dox:68]: I wrote two sentences in the comment and refer to the according section in the TE spec.

File: doc/spec_exceptionHandling.dox [hunk 2]
```diff
@@ -175,7 +184,7 @@ Note: This section defines the internal interface on a low level. Helper functio
 \subsubsection spec_execptionHandling_high_level_implmentation_decorator_behaviour Behaviour
 - \anchor c_2_2 2.2 In doPreWrite() the RecoveryHelper is updated while holding a shared lock on DeviceModule::recoveryMutex:
   - \anchor c_2_2_1 2.2.1 These steps need to be done unconditionally at the very beginning of doPreWrite(), before \ref c_2_4 "2.4" and before delegating to preWrite(). \ref comment_c_2_2_1 "(*)"
-  - 2.2.2 If the RecoveryHelper::wasWritten flag was previously not set, the return value of doWriteTransfer() must be forced to true (data lost).
+  - 2.2.2 If the RecoveryHelper::wasWritten flag was previously not set and the writeOrder is not 0 any more, the return value of doWriteTransfer() must be forced to true (data lost).
```
[the remark this answers is not in the document]
Comment (killenb) [reply doc/spec_exceptionHandling.dox:187]: I added two sentences in 2.1.2, where the write order is introduced.

File: doc/spec_exceptionHandling.dox [hunk 3]
```diff
@@ -1,9 +1,11 @@
 // put the namespace around the doxygen block so we don't have to give it all the time in the code to get links
 namespace ChimeraTK {
 /**
-\page spec_execptionHandling Technical specification: Exception handling for device runtime errors
+\page spec_execptionHandling Technical specification: Exception handling for device runtime errors V1.0RC1WIP
 
-<b>DRAFT VERSION, WRITE-UP IN PROGRESS!</b>
+> **This is a release candidate in implementation. The official V1.0 release will be done once the implementation is ready and we know that the specified behavious is working as intended.**
```
Comment (vargheseg) [inline doc/spec_exceptionHandling.dox:6]: typo: behavious ->behaviour

File: doc/spec_exceptionHandling.dox [hunk 4]
```diff
@@ -43,7 +46,7 @@ namespace ChimeraTK {
   - \anchor exceptionHandling_b_2_1 2.1 The exception status is published as a process variable together with an error message. [\ref testExceptionHandling_b_2_1 "T"]
     - 2.1.1 The variable \c Devices/\<alias\>/status contains a boolean flag whether the device is in an error state.
     - 2.1.2 The variable \c Devices/\<alias\>/message contains an error message, if the device is in an error state, or an empty string otherwise.
-  - \anchor b_2_2 2.2 Read operations will propagate the DataValidity::faulty flag to the owning module / fan out (without changing the actual data value of the process variable):
+  - \anchor b_2_2 2.2 Read operations will propagate the DataValidity::faulty flag to the owning module / fan out:
```
Comment (vargheseg) [inline doc/spec_exceptionHandling.dox:49]: Is it more appropriate for the data validity propagation spec to test this?

File: doc/spec_exceptionHandling.dox (hunk 3 above)
Comment (killenb) [reply doc/spec_exceptionHandling.dox]: fixed

File: doc/spec_exceptionHandling.dox (hunk 4 above)
Comment (killenb) [reply doc/spec_exceptionHandling.dox:49]: Yes, the "propagation" part is tested there. However, that spec is very brief.This point is also about the ExceptionHandlingDecorator changing the flag. And this needs to be tested. The data validity propagation spec refers to this spec about it and does not specify anything where the changing flags are coming from. Maybe we there refer to this test. From the context it fit better here.

**Pull request merged**